File: cube_app/scripts/generate-app-icons.js
/**
 * 从 SVG 生成 Android 应用图标各尺寸 PNG
 * 用法: node scripts/generate-app-icons.js
 */
const fs = require('fs');
const path = require('path');
const { chromium } = require('playwright');

const svgPath = path.resolve(__dirname, '../../cube_front/src/assets/cube.svg');
const resDir = path.resolve(__dirname, '../android/app/src/main/res');

const sizes = {
  'mipmap-mdpi': 48,
  'mipmap-hdpi': 72,
  'mipmap-xhdpi': 96,
  'mipmap-xxhdpi': 144,
  'mipmap-xxxhdpi': 192,
};

async function main() {
  const svgContent = fs.readFileSync(svgPath, 'utf-8');
  const browser = await chromium.launch();
  const page = await browser.newPage();

  for (const [folder, size] of Object.entries(sizes)) {
    const folderPath = path.join(resDir, folder);
    if (!fs.existsSync(folderPath)) fs.mkdirSync(folderPath, { recursive: true });

    // 生成带背景的方形图标
    const html = `
      <div style="width:${size}px;height:${size}px;background:#ffffff;display:flex;align-items:center;justify-content:center;">
        ${svgContent.replace('width="200"', `width="${Math.floor(size * 0.75)}"`).replace('height="200"', `height="${Math.floor(size * 0.75)}"`)}
      </div>`;
    await page.setContent(html, { waitUntil: 'networkidle' });
    const el = await page.$('div');
    await el.screenshot({ path: path.join(folderPath, 'ic_launcher.png'), type: 'png' });

    // 生成圆形图标
    const htmlRound = `
      <div style="width:${size}px;height:${size}px;background:#ffffff;border-radius:50%;display:flex;align-items:center;justify-content:center;overflow:hidden;">
        ${svgContent.replace('width="200"', `width="${Math.floor(size * 0.65)}"`).replace('height="200"', `height="${Math.floor(size * 0.65)}"`)}
      </div>`;
    await page.setContent(htmlRound, { waitUntil: 'networkidle' });
    const elRound = await page.$('div');
    await elRound.screenshot({ path: path.join(folderPath, 'ic_launcher_round.png'), type: 'png' });

    // 生成前景图（adaptive icon）
    const htmlFg = `
      <div style="width:${size}px;height:${size}px;display:flex;align-items:center;justify-content:center;">
        ${svgContent.replace('width="200"', `width="${Math.floor(size * 0.7)}"`).replace('height="200"', `height="${Math.floor(size * 0.7)}"`)}
      </div>`;
    await page.setContent(htmlFg, { waitUntil: 'networkidle' });
    const elFg = await page.$('div');
    await elFg.screenshot({ path: path.join(folderPath, 'ic_launcher_foreground.png'), type: 'png', omitBackground: true });

    console.log(`✓ ${folder} (${size}px)`);
  }

  await browser.close();
  console.log('\n全部图标生成完成');
}

main().catch(e => { console.error(e); process.exit(1); });

File: cube_app/scripts/generate_app_icons.py
"""
从 cube.svg 生成 Android 应用图标各尺寸 PNG
"""
import os
from pathlib import Path
from playwright.sync_api import sync_playwright

BASE_DIR = Path(__file__).resolve().parent.parent
SVG_PATH = BASE_DIR.parent / "cube_front" / "src" / "assets" / "cube.svg"
RES_DIR = BASE_DIR / "android" / "app" / "src" / "main" / "res"

SIZES = {
    "mipmap-mdpi": 48,
    "mipmap-hdpi": 72,
    "mipmap-xhdpi": 96,
    "mipmap-xxhdpi": 144,
    "mipmap-xxxhdpi": 192,
}

svg_content = SVG_PATH.read_text(encoding="utf-8")

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()

    for folder, size in SIZES.items():
        folder_path = RES_DIR / folder
        folder_path.mkdir(parents=True, exist_ok=True)

        icon_size = int(size * 0.72)

        # 方形图标（白色背景）
        html = f"""
        <div style="width:{size}px;height:{size}px;background:#ffffff;display:flex;align-items:center;justify-content:center;">
            {svg_content.replace('width="200"', f'width="{icon_size}"').replace('height="200"', f'height="{icon_size}"')}
        </div>"""
        page.set_content(html)
        div = page.query_selector("div")
        div.screenshot(path=str(folder_path / "ic_launcher.png"), type="png")

        # 圆形图标
        html_round = f"""
        <div style="width:{size}px;height:{size}px;background:#ffffff;border-radius:50%;display:flex;align-items:center;justify-content:center;overflow:hidden;">
            {svg_content.replace('width="200"', f'width="{int(size * 0.62)}"').replace('height="200"', f'height="{int(size * 0.62)}"')}
        </div>"""
        page.set_content(html_round)
        div = page.query_selector("div")
        div.screenshot(path=str(folder_path / "ic_launcher_round.png"), type="png")

        # 前景图（adaptive icon，透明背景）
        html_fg = f"""
        <div style="width:{size}px;height:{size}px;display:flex;align-items:center;justify-content:center;">
            {svg_content.replace('width="200"', f'width="{int(size * 0.75)}"').replace('height="200"', f'height="{int(size * 0.75)}"')}
        </div>"""
        page.set_content(html_fg)
        div = page.query_selector("div")
        div.screenshot(path=str(folder_path / "ic_launcher_foreground.png"), type="png", omit_background=True)

        print(f"✓ {folder} ({size}px)")

    browser.close()

print("\n全部图标生成完成")
